libsolidity/formal/SMTChecker.cpp

```diff
@@ -193,13 +206,19 @@ bool SMTChecker::visit(ForStatement const& _node)
 	_node.body().accept(*this);
 	if (_node.loopExpression())
 		_node.loopExpression()->accept(*this);
-
 	m_interface->pop();
 
-	m_loopExecutionHappened = true;
+	auto indicesAfterLoop = copyVariableIndices();
+	// We reset the execution to before the loop
```

Comment:
Can you add a comment about unrolling? You explain what is done but not why.

Reply:
The unrolling part is the same as before: we run it once. The difference is that before all the variables touched inside the loop body were reset after the loop, whereas now it merges with the branch that doesn't go into the loop, keeping the branch's knowledge

Reply:
Will add a comment